x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -134,25 +142,59 @@ export class ActionsClient {
       actionTypeId: result.attributes.actionTypeId as string,
       name: result.attributes.name as string,
       config: result.attributes.config as Record<string, any>,
+      isPreconfigured: false,
     };
   }
 
   /**
    * Get an action
    */
   public async get({ id }: { id: string }): Promise<ActionResult> {
+    const preconfiguredConnector = this.preconfiguredConnectors.find(
+      pConnector => pConnector.id === id
+    );
+    if (preconfiguredConnector !== undefined) {
+      return {
+        id,
+        actionTypeId: preconfiguredConnector.actionTypeId,
+        name: preconfiguredConnector.name,
+        config: preconfiguredConnector.config,
+        description: preconfiguredConnector.description,
+        isPreconfigured: true,
+      };
+    }
     const result = await this.savedObjectsClient.get<RawAction>('action', id);
 
     return {
       id,
       actionTypeId: result.attributes.actionTypeId,
       name: result.attributes.name,
       config: result.attributes.config,
+      isPreconfigured: false,
     };
   }
 
   /**
-   * Find actions
+   * Get all actions connectors with preconfigured connectors
+   */
+  public async getAll(): Promise<FindActionResult[]> {
+    const savedObjectsActions = (
+      await this.savedObjectsClient.find<RawAction>({
+        perPage: 10000,
+        type: 'action',
+      })
+    ).saved_objects.map(actionFromSavedObject);
+
+    const mergedResult = [...savedObjectsActions, ...this.preconfiguredConnectors];
```

Comment:
Should we provide a default sort on these so the response doesn't seem like we tacked on the preconfigured actions? Something like `array.sort((a, b) => a.name.localeCompare(b.name));`

Reply:
good point - will add a default sorting